src/hooks/useTikTokAuth.ts

```diff
@@ -24,9 +26,23 @@ type WithAuthCallback = (accessToken: string, openId: string) => void
  * @returns {Function}
  */
 export const useTikTokAuth = ({
-  onError
+  onError: errorCallback
 }: useTikTokAuthOptions): ((callback: WithAuthCallback) => void) => {
-  const withAuth = (callback: WithAuthCallback) => {
+  const record = useRecord()
+
+  const onError = (e: Error) => {
+    record(make(Name.TIKTOK_OAUTH_ERROR, { error: e.message }))
+    errorCallback(e)
+  }
+
+  const withAuth = (successCallback: WithAuthCallback) => {
+    record(make(Name.TIKTOK_START_OAUTH, {}))
+
+    const callback: WithAuthCallback = (accessToken, openId) => {
+      record(make(Name.TIKTOK_COMPLETE_OAUTH, {}))
```

Comment:
nit, probably is better to record after invoking the callback (esp if we don't wait for the callback) in the off chance that this could somehow throw

Reply:
Agreed, fixed!